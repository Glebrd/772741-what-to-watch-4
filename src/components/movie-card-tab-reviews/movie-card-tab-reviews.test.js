import React from "react";
import renderer from "react-test-renderer";
import MovieCardTabReviews from "./movie-card-tab-reviews";

const movie = {
  id: 124,
  title: `Se7en`,
  picture: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Thriller`,
  date: `1995`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  rating: 7.9,
  scores: 685,
  director: `Chic Alvar`,
  starring: [`Padraic Wetheril`, `Gnni Fenkel`, `Ichabod Kuhlen`, `Andy MacRanald`],
  description: `Two detectives, a rookie and a veteran, hunt a serial killer who uses the seven deadly sins as his motives.`,
  videoPreview: `https: //upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  runTime: `2h 07m`,
  comments: [{
    id: 1,
    user: {
      id: 562,
      name: `Leila Boswood`,
    },
    rating: 9,
    comment: ` Se7en is well crafted and ingeniously clever, making it one of the greatest films of the 90's`,
    date: `March 16, 2010`
  }, {
    id: 2,
    user: {
      id: 248,
      name: `Melvin Dutteridge`,
    },
    rating: 8,
    comment: `Dark and disturbing thriller that will stay with you forever.`,
    date: `July 22, 2015`
  }, {
    id: 3,
    user: {
      id: 182,
      name: `Ely Ernke`,
    },
    rating: 10,
    comment: `One of the best-made films of its era and genre`,
    date: `November 10, 2008`
  }]
};

const {comments} = movie;

it(`Movie card tab reviews renders correctly`, () => {
  const tree = renderer
    .create(
        <MovieCardTabReviews
          comments={comments}
        />).toJSON();

  expect(tree).toMatchSnapshot();
});
