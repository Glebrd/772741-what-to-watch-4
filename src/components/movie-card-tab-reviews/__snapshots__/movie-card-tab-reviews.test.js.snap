// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card tab reviews renders correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
           Se7en is well crafted and ingeniously clever, making it one of the greatest films of the 90's
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2015-11-18"
          >
            March 16, 2010
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Dark and disturbing thriller that will stay with you forever.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2015-11-18"
          >
            July 22, 2015
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          One of the best-made films of its era and genre
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2015-11-18"
          >
            November 10, 2008
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        10
      </div>
    </div>
  </div>
</div>
`;
